cmd/verify/verify.go

```diff
@@ -152,42 +130,49 @@ func verify(cmd *cobra.Command, files []string) {
 		recordReaderWg.Wait()
 
 		if recordCount == 0 {
-			logger.Error("no record in file", "file", filepath)
+			slog.Error("no record in file", "file", filepath)
 		}
 
 		// Ensure there is a visible difference when errors are present.
 		if errorsCount > 0 {
-			logger.Error(fmt.Sprintf("checked in %s", time.Since(startTime).String()), "file", filepath, "valid", valid, "errors", errorsCount, "count", recordCount, "allRecordsRead", allRecordsRead)
+			slog.Error(fmt.Sprintf("checked in %s", time.Since(startTime).String()), "file", filepath, "valid", valid, "errors", errorsCount, "count", recordCount, "allRecordsRead", allRecordsRead)
 		} else {
-			logger.Info(fmt.Sprintf("checked in %s", time.Since(startTime).String()), "file", filepath, "valid", valid, "errors", errorsCount, "count", recordCount, "allRecordsRead", allRecordsRead)
+			slog.Info(fmt.Sprintf("checked in %s", time.Since(startTime).String()), "file", filepath, "valid", valid, "errors", errorsCount, "count", recordCount, "allRecordsRead", allRecordsRead)
 		}
 
 	}
 }
 
+func processVerifyRecord(record *warc.Record, filepath string, results chan<- result) {
+	var res result
+	res.blockDigestErrorsCount, res.blockDigestValid = verifyBlockDigest(record, filepath)
+	res.payloadDigestErrorsCount, res.payloadDigestValid = verifyPayloadDigest(record, filepath)
+	res.warcVersionValid = verifyWARCVersion(record, filepath)
+	results <- res
+}
+
 func verifyPayloadDigest(record *warc.Record, filepath string) (errorsCount int, valid bool) {
 	valid = true
 
-	// Verify that the Payload-Digest field exists
+	// WARC-Payload-Digest is optional in both WARC 1.0 and 1.1 specifications
+	// If it's not present, that's perfectly valid - just return success
 	if record.Header.Get("WARC-Payload-Digest") == "" {
-		logger.Error("WARC-Payload-Digest is missing", "file", filepath, "recordID", record.Header.Get("WARC-Record-ID"))
-		valid = false
-		errorsCount++
+		slog.Debug("WARC-Payload-Digest not present (this is optional)", "file", filepath, "recordID", record.Header.Get("WARC-Record-ID"))
```

Comment:
maybe not the place, but it could be nice to add that it is certainly recommended for deduplication purposes.